aten/src/ATen/native/Histogram.cpp

```diff
@@ -6,117 +6,181 @@
  *
  * Restrictions are defined in histogram_check_inputs() and in select_outer_bin_edges().
  */
 
 namespace at { namespace native {
 
-DEFINE_DISPATCH(histogram_stub);
-
-DEFINE_DISPATCH(histogram_linear_stub);
+DEFINE_DISPATCH(histogramdd_stub);
+DEFINE_DISPATCH(histogramdd_linear_stub);
 
 namespace {
 
 /* Checks properties of input tensors input, bins, and weight.
  */
-void histogram_check_inputs(const Tensor& input, const Tensor& bins, const c10::optional<Tensor>& weight) {
-    TORCH_CHECK(input.dtype() == bins.dtype(), "torch.histogram: input tensor and bins tensor should",
-            " have the same dtype, but got input ", input.dtype(), " and bins ", bins.dtype());
+void histogramdd_check_inputs(const Tensor& input, const TensorList& bins, const c10::optional<Tensor>& weight) {
+    TORCH_CHECK(input.dim() >= 2, "torch.histogramdd: input tensor should have at least 2 dimensions");
+
+    const int64_t D = input.size(-1);
+
+    TORCH_CHECK(int64_t(bins.size()) == D, "torch.histogramdd: expected ", D, " sequences of bin edges for a ", D,
+                "-dimensional histogram but got ", bins.size());
 
-    TORCH_CHECK(bins.dim() == 1, "torch.histogram: bins tensor should have dimension 1,",
-            " but got ", bins.dim(), " dimension");
+    for (int64_t dim = 0; dim < D; dim++) {
+        TORCH_CHECK(input.dtype() == bins[dim].dtype(), "torch.histogramdd: input tensor and bins tensor should",
+                " have the same dtype, but got input ", input.dtype(),
+                " and bins for dimension ", dim, bins[dim].dtype());
 
-    TORCH_CHECK(bins.numel() > 0, "torch.histogram: bins tensor should have at least 1 element,",
-            " but got ", bins.numel(), " elements");
+        TORCH_CHECK(bins[dim].dim() == 1, "torch.histogramdd: bins tensor should have dimension 1,",
+                " but got ", bins[dim].dim(), " dimensions for dimension ", dim);
+
+        TORCH_CHECK(bins[dim].numel() > 0, "torch.histogramdd: bins tensor should have at least 1 element,",
+                " but got ", bins[dim].numel(), " elements for dimension ", dim);
+    }
 
     if (weight.has_value()) {
         TORCH_CHECK(input.dtype() == weight.value().dtype(), "torch.histogram: if weight tensor is provided,"
                 " input tensor and weight tensor should have the same dtype, but got input(", input.dtype(), ")",
                 ", and weight(", weight.value().dtype(), ")");
 
-        TORCH_CHECK(input.sizes() == weight.value().sizes(), "torch.histogram: if weight tensor is provided,"
-                " input tensor and weight tensor should have the same shape, but got input(", input.sizes(), ")",
-                ", and weight(", weight.value().sizes(), ")");
+        auto input_sizes = input.sizes().vec();
+        input_sizes.pop_back();
+
+        auto weight_sizes = weight.value().sizes().vec();
+        if (weight_sizes.empty())
```

Comment:
Internal lint will want curly braces for every if statement, including one-line if statements